src/components/sortableGridList/SortableItem.tsx

```diff
@@ -40,24 +38,47 @@ function SortableItem(props: PropsWithChildren<SortableItemProps>) {
   const tempTranslateX = useSharedValue(0);
   const tempTranslateY = useSharedValue(0);
 
+  const dataManuallyChanged = useSharedValue(false);
+
+  useDidUpdate(() => {
+    dataManuallyChanged.value = true;
+    initialIndex.value = _.map(data, 'id').indexOf(id);
+  }, [data]);
+
+  useAnimatedReaction(() => itemsOrder.value,
+    (currItemsOrder, prevItemsOrder) => {
+      // Note: Unfortunately itemsOrder sharedValue is being initialized on each render
+      // Therefore I added this extra check here that compares current and previous values
+      // See open issue: https://github.com/software-mansion/react-native-reanimated/issues/3224
+      if (prevItemsOrder === null || currItemsOrder.join(',') === prevItemsOrder.join(',')) {
+        return;
+      } else {
+        const newOrder = getItemOrderById(currItemsOrder, id);
+        const prevOrder = getItemOrderById(prevItemsOrder, id);
+
+        if (newOrder === initialIndex.value) {
+          if (dataManuallyChanged.value) {
+            translateX.value = 0;
+            translateY.value = 0;
+          } else {
+            translateX.value = withTiming(0, animationConfig);
+            translateY.value = withTiming(0, animationConfig);
+          }
+          dataManuallyChanged.value = false;
+        } else if (newOrder !== prevOrder) {
+          const translation = getTranslationByOrderChange(newOrder, prevOrder);
+          translateX.value = withTiming(translateX.value + translation.x, animationConfig);
+          translateY.value = withTiming(translateY.value + translation.y, animationConfig);
+        }
+      }
```

Comment:
I know we don't use to do it, but I found the conditions here a bit confusing.. consider adding a comment for every condition that explains the cases that are relevant to this condition..

Reply:
Done